sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -260,10 +259,39 @@ export class Recorder {
         logger.verbose("[Recorder#start] Setting redirect mode");
         await setRecordingOptions(Recorder.url, this.httpClient, { handleRedirects: !isNode });
         logger.verbose("[Recorder#start] Sending the start request to the test proxy");
-        const rsp = await this.httpClient.sendRequest({
+        let rsp = await this.httpClient.sendRequest({
           ...req,
           allowInsecureConnection: true,
         });
+
+        // If the error is due to the assets.json not existing, try again without specifying an assets.json. This will
+        // occur with SDKs that have not migrated to asset sync yet.
+        // TODO: remove once everyone has migrated to asset sync
```

Comment:
Log an issue to track this removal too please.

Reply:
#24792